components/datetime/src/pattern/reference/parser.rs

```diff
@@ -128,9 +128,11 @@ impl<'p> Parser<'p> {
                 return Err(PatternError::UnclosedLiteral);
             }
             Segment::Literal { literal, .. } => {
-                for ch in literal.chars() {
-                    result.push(ch.into());
-                }
+                let items = literal
+                    .chars()
+                    .map(|ch| ch.into())
+                    .collect::<Vec<PatternItem>>();
+                result.extend(items);
```

Comment:
`question`: I'm concerned about this code. If I'm not mistaken you're allocating a new temporary `Vec` that you collect into, and then extend `result` with it.

The previous code did not allocate.

What I'd suggest to do is `result.extend(literal.chars().map(|ch| ch.into()))` - this way you're passing an iterator to extend, not an allocated vector.

Reply:
Thanks, that's fair observation and I wanted to ask about it during review :) I initially tried to implement it exactly as you proposed however it results in type inference error:

```bash
error[E0282]: type annotations needed
   --> components/datetime/src/pattern/reference/parser.rs:151:28
    |
151 |                     result.extend(literal.chars().map(|ch| ch.into()))
    |                            ^^^^^^                          --------- this method call resolves to `T`
    |                            |
    |                            cannot infer type for type parameter `A` declared on the trait `Extend`
```

Do you have any suggestion on how to specify type somewhere so that it can work?

Reply:
Ah, I think I could use `result.extend(literal.chars().map(GenericPatternItem::from))` to hint on the type. Does that look good?

Reply:
Fixed 👍